service/resource/configmap/configmap_test.go

```diff
@@ -1,3 +1,204 @@
 package configmap
 
-// TODO: Test New()
+import (
+	"testing"
+	"time"
+
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+
+	"github.com/giantswarm/prometheus-config-controller/service/prometheus"
+)
+
+// Test_Resource_Configmap_New tests the New function.
+func Test_Resource_Configmap_New(t *testing.T) {
+	prometheusReloaderConfig := prometheus.DefaultConfig()
+
+	prometheusReloaderConfig.K8sClient = fake.NewSimpleClientset()
+	prometheusReloaderConfig.Logger = microloggertest.New()
+
+	prometheusReloaderConfig.Address = "http://127.0.0.1:9090"
+	prometheusReloaderConfig.ConfigMapKey = "prometheus.yml"
+	prometheusReloaderConfig.ConfigMapName = "prometheus"
+	prometheusReloaderConfig.ConfigMapNamespace = "monitoring"
+	prometheusReloaderConfig.MinimumReloadTime = 2 * time.Minute
+
+	prometheusReloader, _ := prometheus.New(prometheusReloaderConfig)
+
+	tests := []struct {
+		config func() Config
+
+		expectedErrorHandler func(error) bool
+	}{
+		// Test that the default config returns an error.
+		{
+			config: DefaultConfig,
```

Comment:
FYI: we're in transition towards removing DefaultConfig function: https://github.com/giantswarm/kvm-operator/pull/352#discussion_r163781057

This happened a moment ago :D

Reply:
Fine, I'll get around to removing here in ~6 months :D

Reply:
:D

Reply:
Eventual consistency.